lib/mail/fields/content_type_field.rb

```diff
@@ -144,10 +144,9 @@ def sanatize( val )
 
       # TODO: check if there are cases where whitespace is not a separator
       val = val.
+        gsub(/\s[\w\-]+:\s.*\\r\\n$/, ''). # cases with botched fields like Content-Type: text/html Content-Transfer-Encoding: 8bit\r\n
```

Comment:
...and what exactly will I do, when an Exception of this sort occurs? Not deliver the email?

Some emails are _constructed_ to work around strict parsers. We have several thousand emails in our regression suite. Broken headers and invalid encodings constitute an important part of that suite.

You don't have to agree with our vision of what this library is supposed to handle. We'll just have to live with a fork.